source/opt/block_merge_util.cpp

```diff
@@ -98,6 +98,17 @@ bool CanMergeWithSuccessor(IRContext* context, BasicBlock* block) {
     return false;
   }
 
+  // Note: This means that the instructions in a break block will execute as if
+  // they were still diverged according to the loop iteration. This restricts
+  // potential transformations an implementation may perform on the IR to match
+  // shader author expectations. Similarly, instructions in the loop construct
+  // cannot be moved into the continue construct unless it can be proven that
+  // invocations are always converged.
+  if (succ_is_merge && context->get_feature_mgr()->HasExtension(
+                           kSPV_KHR_maximal_reconvergence)) {
```

Comment:
I wonder if we should do this all of the time. Is there a reason not to?

Reply:
Maybe we should disallow merging with unresolve comments so auto-merge don't betrays us.

Well, the main reason is it is not disallowed by the spec. So I wanted to be conservative in the changes we do.